### PR TITLE
Ifpack2 - BlockTriDiags team policy only.

### PR DESCRIPTION
**Pull request by kyungjoo-kim**
## Description

Previously, the code separates RangePolicy for host architecture from TeamPolicy for gpu architectures. This makes code difficult to read and we can achieve a single version of code with proper meta programming. This PR includes the modification for the team policy only version.

## Motivation and Context

While I am working on the code, it is too much headache working on two codes for two architectures. 

## How Has This Been Tested?

Tested from white and bowman.

### CODE REVIEW

File: packages/ifpack2/test/unit_tests/Ifpack2_UnitTestBlockTriDiContainerUtil.hpp [hunk 1]
```diff
@@ -246,7 +246,7 @@ struct BlockTriDiContainerTester {
           auto input = T_bare->createDefaultApplyParameters();
           input.zeroStartingSolution = zero_starting;
           input.maxNumSweeps = num_sweeps;
-          input.tolerance = norm_based > 1 ? tol : 0;
+          input.tolerance = norm_based ? tol : 0;
```
Comment (mhoemmen) [inline packages/ifpack2/test/unit_tests/Ifpack2_UnitTestBlockTriDiContainerUtil.hpp]: Didn't we handle this in a separate PR?

File: packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp [hunk 2]
```diff
@@ -247,11 +247,12 @@ namespace Ifpack2 {
       template<typename T, int l> using Vector = KokkosBatched::Experimental::Vector<T,l>;
       template<typename T> using SIMD = KokkosBatched::Experimental::SIMD<T>;
       template<typename T, typename M> using DefaultVectorLength = KokkosBatched::Experimental::DefaultVectorLength<T,M>;
+      template<typename T, typename M> using DefaultInternalVectorLength = KokkosBatched::Experimental::DefaultInternalVectorLength<T,M>;
```
Comment (mhoemmen) [inline packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp]: If the alias has the same number of template parameters as the original, you could just say `using KokkosBatched::Experimental::DefaultInternalVectorLength;`.  This is not incorrect, though.

File: packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp [hunk 3]
```diff
@@ -1065,30 +1069,15 @@ namespace Ifpack2 {
             const local_ordinal_type iend = pack_td_ptr(packptr(k+1));
             const local_ordinal_type diff = iend - ibeg;
             const local_ordinal_type icount = diff/3 + (diff%3 > 0);
-	    Kokkos::parallel_for(Kokkos::TeamThreadRange(member,icount),[&](const local_ordinal_type &ii) {
-		Kokkos::parallel_for(Kokkos::ThreadVectorRange(member, vector_length_value),[&](const int &v) {
+            Kokkos::parallel_for(Kokkos::ThreadVectorRange(member, vector_loop_size),[&](const int &v) {
+                Kokkos::parallel_for(Kokkos::TeamThreadRange(member,icount),[&](const local_ordinal_type &ii) {
                     const local_ordinal_type i = ibeg + ii*3;
                     for (local_ordinal_type j=0;j<blocksize;++j) 
                       values(i,j,j,v) = 1;
```
Comment (mhoemmen) [inline packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp]: Do we build this with complex Scalar types?  If so, I don't think this assignment (from an integer `1` to `impl_scalar_type`, which could be `Kokkos::complex<double>` or `Kokkos::complex<float>`) will compile.  The easy fix is to change the number to `1.0`.

File: packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp [hunk 4]
```diff
@@ -1424,14 +1413,35 @@ namespace Ifpack2 {
     ///
     /// numeric phase, initialize the preconditioner
     ///
+    template<typename ArgActiveExecutionMemorySpace>
+    struct EtxtractAndFactorizeTridiagsDefaultModeAndAlgo;
```
Comment (mhoemmen) [inline packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp]: Spelling: "ExtractAnd..."

File: packages/ifpack2/test/unit_tests/Ifpack2_UnitTestBlockTriDiContainerUtil.hpp (hunk 1 above)
Comment (kyungjoo-kim) [reply packages/ifpack2/test/unit_tests/Ifpack2_UnitTestBlockTriDiContainerUtil.hpp]: Yes. I include this for a testing purpose as the Andrew's PR is not yet merged. Once his PR is merged and I will include a merge commit.

File: packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp (hunk 2 above)
Comment (kyungjoo-kim) [reply packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp]: I use this aliasing syntax after I experience a problem from a certain compiler a long time ago. Probably the problem is gone now. It is just my habit. I am working on improvement and preparing for the second PR and I will test the simple aliasing syntax and include the modification.

File: packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp (hunk 3 above)
Comment (kyungjoo-kim) [reply packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp]: This is okay even for complex. Although nobody uses this with complex but I tested with complex before submitting PR.

File: packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp (hunk 4 above)
Comment (kyungjoo-kim) [reply packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp]: Thanks. This should be fixed now.

File: packages/ifpack2/test/unit_tests/Ifpack2_UnitTestBlockTriDiContainerUtil.hpp (hunk 1 above)
Comment (mhoemmen) [reply packages/ifpack2/test/unit_tests/Ifpack2_UnitTestBlockTriDiContainerUtil.hpp]: FYI, that PR looks like it's making the BlockTriDi test fail.

**Comment by kyungjoo-kim** [comment]
@mhoemmen would you approve one more time ?

**Pull request merged**